File: git-reword-commit-cli/bindings/java/src/com/gitreword/Platform.java
package com.gitreword;

/**
 * Platform detection utilities for git-reword.
 */
public class Platform {
    static {
        NativeLibrary.load();
    }
    
    public static native boolean isWindows();
    public static native boolean isMacOS();
    public static native boolean isLinux();
    public static native boolean isAndroid();
    public static native boolean isIOS();
    public static native boolean isTermux();
    public static native String getPlatformName();
}

File: git-reword-commit-cli/bindings/dart/lib/src/platform.dart
part of '../git_reword.dart';

/// Platform detection utilities.
class GitRewordPlatform {
  GitRewordPlatform._();

  /// Check if running on Windows.
  static bool get isWindows => _NativeBindings.platformIsWindows() != 0;

  /// Check if running on macOS.
  static bool get isMacOS => _NativeBindings.platformIsMacOS() != 0;

  /// Check if running on Linux.
  static bool get isLinux => _NativeBindings.platformIsLinux() != 0;

  /// Check if running on Android.
  static bool get isAndroid => _NativeBindings.platformIsAndroid() != 0;

  /// Check if running on iOS.
  static bool get isIOS => _NativeBindings.platformIsIOS() != 0;

  /// Check if running in Termux environment.
  static bool get isTermux => _NativeBindings.platformIsTermux() != 0;

  /// Get the platform name.
  static String get name {
    final ptr = _NativeBindings.platformGetName();
    final result = ptr.toDartString();
    _NativeBindings.freeString(ptr);
    return result;
  }
}
